sphinxext/template.py

```diff
@@ -21,7 +21,8 @@
 
 PAT_SUBSTITUTION = re.compile(r'^\$[\w\.]+$')
 PAT_RST_SECTION = re.compile(r'(.*)\n((?:^----+$)|(?:^====+$)|(?:^~~~~+$)|(?:^````+$))', re.M)
-
+LANGUAGES = ['shell', 'python', 'java', 'nodejs', 'php', 'c', 'cpp11', 'csharp', 'perl', 'ruby', 'scala', 'motor', 'go']
```

Comment:
It probably makes sense to encode these as one list of (str, str) tuples to help make sure they stay in the right order.